src/components/Web3Status/index.tsx

```diff
@@ -206,10 +208,12 @@ function Web3StatusInner() {
   const theme = useTheme()
   const toggleWalletDropdown = useToggleWalletDropdown()
   const toggleWalletModal = useToggleWalletModal()
+  const toggleMetamaskConnectionErrorModal = useToggleMetamaskConnectionErrorModal()
   const walletIsOpen = useModalIsOpen(ApplicationModal.WALLET_DROPDOWN)
   const isClaimAvailable = useIsNftClaimAvailable((state) => state.isClaimAvailable)
 
   const error = useAppSelector((state) => state.connection.errorByConnectionType[getConnection(connector).type])
+  if (getIsMetaMask() && error) toggleMetamaskConnectionErrorModal()
```

Comment:
This will toggle, so this code may flicker the modal. You should be putting this in a `useEffect`, and only invoking it when there is an error.

Reply:
not sure what u mean by the toggle may flicker the modal? I wrapped it in a useEffect regardless.